planetary-social/nos
# Add Delete confirmation  before deleting account

## Issues covered
Last part of #80 

## Description
I started with using the `AlertState` used widely within the app, then I found out that it doesn't support adding textfields.
I then moved to using the `.alert` with a textfield incorporated in iOS 17. I implemented this already but found a weird bug where an initially disabled button in the alert view  would not update its state immediately after it has been enabled, until the user deletes the text and types it again. I found people experiencing something similar [here](https://forums.developer.apple.com/forums/thread/737964), and realised it was an iOS bug 🤦‍♀️.

I then finally created a custom alert view with the textfield and overlay it on the `SettingsView`'s main content with a `ZStack` when necessary.
@mplorentz  has implemented the majority of the work for this feature. This is just the part where the user types "DELETE" before actually deleting the account. I discussed with @setch-l yesterday about the multi-language support for deletion and we concluded on using just "DELETE" instead of "DELETE ACCOUNT" because of languages like German, etc with large amount of characters for a word.

Figma: https://www.figma.com/design/s0qf4VmyQygydP8MIQazZc/Nos---Dev-Ready?node-id=8661-6918&t=cCImyKBTjkMyVOvg-4

## How to test
1. Build the code.
2. Navigate to Settings screen.
3. Scroll to the bottom of the screen and click the delete my account button.
4. Please check that:
    - The new alert view with a textfield overlays the SettingsView content.
    - The delete button is not enabled until the user types exactly "DELETE" in their respective language enabled on their                device.
## Screenshots/Video
Before:
<img width="713" alt="Screenshot 2024-10-16 at 1 23 08 PM" src="https://github.com/user-attachments/assets/bff4ffdf-d96b-4029-bb9f-2ff0999d1694">

After:
English

https://github.com/user-attachments/assets/5d833d7f-55c2-4e7a-9f1e-1505dfa3a21d

Spanish

https://github.com/user-attachments/assets/dd4369c4-1322-44ad-af27-42a6e6f3f17a
